modules/core/src/main/scala/org/tessellation/infrastructure/rewards/Rewards.scala

```diff
@@ -74,10 +73,19 @@ object Rewards {
 
           val mintedAmount = getAmountByEpoch(epochProgress, config.rewardsPerEpoch)
 
-          allRewardsState
+          val rewards: F[SortedSet[RewardTransaction]] = allRewardsState
             .run(mintedAmount)
             .flatTap(validateState(mintedAmount))
             .map(toTransactions)
+
+          val oneTimeRewards = config.oneTimeRewards
+            .filter(_.epoch === epochProgress)
+            .map(otr => RewardTransaction(otr.address, otr.txnAmount))
```

Comment:
the issue I see is that you cannot configure one time reward and launch it the very same moment you release the filtering out feature. you need to lock the address first and change the reward logic, then read the balance of stardust wallet and configure one time reward, then release again with this one time reward (happening at some higher epoch progress). take a look at my draft PR - I create the reward transaction during snapshot creation process when I read the current balance and it's executed also one time at a specific epoch progress.

Reply:
> you need to lock the address first and change the reward logic
It's unspendable so it can't change here.
>  then read the balance of stardust wallet and configure one time reward
We're just getting the balance and directly encoding it as a one time reward based on current block explorer to avoid any check on the balances, for any delta associated with the 4 day interim we can handle that separately.

Reply:
> It's unspendable so it can't change here.
yes, but the reward logic should be changed to stop sending rewards to that address

> We're just getting the balance and directly encoding it as a one time reward based on current block explorer to avoid any check on the balances, for any delta associated with the 4 day interim we can handle that separately.

so effectively you change the total supply for few days, right? is that acceptable?

Reply:
@ryle-goehausen  I have to say I agree w/ @marcinwadon here. We need this balance transferred cleanly in a single (reward) transaction of the full amount. We shouldn't hardcode a value that leaves a remainder that will need to be cleaned up later. This is an unusual change for the network so it's important it's extremely clear what we've done. 

So if I'm understanding correctly, the options would be either a single change that switches Stardust rewards address and triggers a reward distribution at a future epochProgress based on a balance read, or else two releases - first switches Stardust rewards address, second hardcodes the final balance for the reward distribution.

Reply:
Or I guess third option would be to calculate an exact amount at a future epochProgress level but we would need to be certain it was correct.

Reply:
@AlexBrandes Would a more precise strategy be to lock the stardust address so it receives no further rewards, then use the balance at an epoch after the lock to transfer to the new stardust?

Reply:
@AlexBrandes hey, I'll see if we can change this to do it exactly. The main reasoning here

> We need this balance transferred cleanly in a single (reward) transaction of the full amount. 

This can be accomplished with the existing change by filling in the value from the balance map at another point in the code. The reason this is a problem is

>We shouldn't hardcode a value 

In order to maintain a validator we'd need to know how to reproduce the calculation of the rewards value, which means we'd need that balance map at all points, the point of hardcoding a value is so that it can be verified independently without a balance map. We'd also need the balance map as of a particular point in time. Let's say someone sends funds to that address afterwards and the balance map changes, we can no longer validate the reward operation as correct. Because it was produced at an exact time, which means now we need to persist the balance map as of that exact block forever in order to validate it.

With a hardcoded value, we don't. So we can get the value from the balance map to make it a single transaction, but then we need to make a code change after the fact that hardcodes this value for the future validation that the reward transaction is correct without maintaining the historical balance map at a point in time. This can be done, but I assumed the other solution was simpler.

Reply:
To clarify even more. It is absolutely not possible to estimate the exact number of rewards in advance, as we can only roughly estimate the block formation time.

Reply:
So again, if the requirement is that the balance transfer happens EXACTLY once, I recommend that we create a one time reward at epoch t+1 after stopping the prior, set it's balance to a 0 value in the configuration, and then inject a code statement which updates it from the balance map (similar to how Marcins PR worked,) then we'll need to replace the config with another PR with the exact value that was transferred so we avoid the validation problem.

Reply:
Summary of discussion from meeting and proposed plan:

We will create two releases. The first release will NOT introduce a one time reward, but instead change the weights associated with existing rewards and migrate the stardust primary to the new address. This updates the case statement and introduces the one time reward logic with an empty list.

The second release will introduce the one time reward with an exact value taken from block explorer for balances from the original stardust primary as of the point in time where they have stopped changing -- providing an exact value.